nav2_system_tests/src/behavior_tree/test_behavior_tree_node.cpp

```diff
@@ -79,10 +79,10 @@ class BehaviorTreeHandler
       return false;
     }
 
-    auto xml_string = std::string(
-      std::istreambuf_iterator<char>(xml_file),
-      std::istreambuf_iterator<char>());
-
+    std::stringstream buffer;
+    buffer << xml_file.rdbuf();
+    xml_file.close();
+    std::string xml_string = buffer.str();
```

Comment:
This achieves the same and avoids the [error](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105580)